File: my-folder/0392-is-subsequence/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int last_index = 0;
        int found = 0;
        bool decision = false;
        for (int i=0;i<s.size();i++)
        {
            bool in_loop=false;
            for (int j=last_index;j<t.size();j++)
            {
                if (t[j] == s[i])
                {
                    last_index = j+1;
                    found++;
                    in_loop=true;
                    break;
                }
            }
            if(!in_loop)
            {
                break;

            }

        }
        if (found>=s.size())
        {
            decision=true;
        }
        return decision;
        
    }
};

File: my-folder/0392-is-subsequence/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s)>len(t):
            return False
        s_counter = Counter(s)
        t_counter = Counter(t)

        for key,val in s_counter.items():
            if key not in t_counter:
                return False
        
        ptr = 0
        for i in range(len(s)):
            while ptr<len(t) and t[ptr]!=s[i]:
                ptr+=1
            if ptr>=len(t) and i<len(s):
                return False
            ptr+=1
        return True

        
        
